cfg/builder/builder_walk.cc

```diff
@@ -822,17 +822,33 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
 
                 for (auto &expr : a.rescueCases) {
                     auto *rescueCase = ast::cast_tree<ast::RescueCase>(expr);
-                    auto caseBody = cctx.inWhat.freshBlock(cctx.loops, handlersRubyRegionId);
+                    auto caseHeader = cctx.inWhat.freshBlock(cctx.loops, handlersRubyRegionId);
                     auto &exceptions = rescueCase->exceptions;
                     auto added = false;
                     auto *local = ast::cast_tree<ast::Local>(rescueCase->var);
                     ENFORCE(local != nullptr, "rescue case var not a local?");
 
                     auto localVar = cctx.inWhat.enterLocal(local->localVariable);
-                    rescueHandlersBlock->exprs.emplace_back(localVar, rescueCase->var.loc(),
-                                                            make_insn<Ident>(exceptionValue));
 
-                    // Mark the exception as handled
+                    // caseBody->exprs.emplace_back(localVar, rescueCase->var.loc(), make_insn<Literal>(core::Types::top()));
+                    auto rescueHeaderTemp = cctx.newTemporary(core::Names::rescueHeaderTemp());
+                    synthesizeExpr(caseHeader, localVar, rescueCase->var.loc(), make_insn<Ident>(exceptionValue));
+                    synthesizeExpr(caseHeader, rescueHeaderTemp, rescueCase->var.loc(), make_insn<Literal>(core::Types::top()));
```

Comment:
This is not allowed. If you want something fake that has type `<top>`, you'll have to emit a call to a magic method that returns `<top>`.